hdaudio-codecs-defconfig

```diff
@@ -5,9 +5,6 @@ CONFIG_SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC=y
 CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC=y
 CONFIG_SND_HDA_INTEL=m
 
-# enable fallback (only SKL for now)
-CONFIG_SND_SOC_INTEL_SKL_LEGACY_SUPPORT=y
-
 # machine driver for HDaudio support
 CONFIG_SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH=m
```

Comment:
@plbossart 
I think for some HDA only platform the following is enough. 
`merge_config.sh .config base-defconfig sof-defconfig hdaudio-codecs-defconfig`

But the`CONFIG_SND_SOC_SOF_HDA_LINK` will not be selected.

Reply:
this is needed for SST compilation. It's harmless to keep.

Reply:
sorry ignore this comment, I replied to the wrong one...

Reply:
you're right, we can add this twice in sof-mach-driver-defconfig and hdaudio-codecs-defconfig. Doesn't hurt to do so.